internal/registration/internal/domain/user_id_test.go

```diff
@@ -0,0 +1,54 @@
+package domain_test
```

Comment:
For each type of constructor except of `NewUserRegistrationID` you should have 2 types of tests - positive and negative. Please add negative test for `Must...` and positive for `Parse...`.